crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -18,6 +18,8 @@ use {
 
 mod tokenized;
 mod trade;
+pub mod transaction;
+pub use transaction::{Transaction, Tx};
```

Comment:
This looks odd. I think either we just expose the transaction module publicly and not reexport anything from it or we only reexport things from it.
Currently you could access these types as `settlement::transaction::Transaction` or `settlement::Transaction` which seems not intended.